### PR TITLE
Improve documentation of unary :

### PR DESCRIPTION
**Pull request by jakobnissen**
Unary `:` was poorly documented previously. The previous docs stated it would
return a Symbol, but in fact also Expr objects and literal values can be
returned from code quoting.

See #43054

### CODE REVIEW

File: base/range.jl [hunk 1]
```diff
@@ -28,14 +28,40 @@ _colon(::Ordered, ::ArithmeticRounds, start::T, step, stop::T) where {T} =
 _colon(::Any, ::Any, start::T, step, stop::T) where {T} =
     StepRangeLen(start, step, floor(Integer, (stop-start)/step)+1)
 
+"""
+	(:)(expr)
+
+`:expr` quotes `expr`, returning an abstract syntax tree (AST) of `expr`.
+The AST may be of type `Expr`, `Symbol`, or a literal value.
+Which of these three types are returned for any given expression is an
```
Comment (KristofferC) [inline base/range.jl:36]: Does this sentence add anything useful?

File: base/range.jl [hunk 2]
```diff
@@ -28,14 +28,40 @@ _colon(::Ordered, ::ArithmeticRounds, start::T, step, stop::T) where {T} =
 _colon(::Any, ::Any, start::T, step, stop::T) where {T} =
     StepRangeLen(start, step, floor(Integer, (stop-start)/step)+1)
 
+"""
```
Comment (fredrikekre) [inline base/range.jl:31]: Should probably be in `base/docs/basedocs.jl` and not `base/range.jl`.

File: base/range.jl (hunk 1 above)
Comment (jakobnissen) [reply base/range.jl:36]: I get what you mean, but part of what I found confusing in the original issue was what type was actually returned.

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -682,6 +682,33 @@ Expr
 """
 Expr
 
+"""
+	(:)(expr)
```
Comment (aviatesk) [inline base/docs/basedocs.jl:686]:
```suggestion
    (:)(expr)
```

**Review by aviatesk — APPROVED** [review]

File: base/docs/basedocs.jl [hunk 4]
```diff
@@ -682,6 +682,33 @@ Expr
 """
 Expr
 
+"""
+    (:)(expr)
+
+`:expr` quotes `expr`, returning an abstract syntax tree (AST) of `expr`.
```
Comment (vchuravy) [inline base/docs/basedocs.jl:688]:
```suggestion
`:expr` quotes the expression `expr`, returning the abstract syntax tree (AST) of `expr`.
```

**Pull request merged**